icu4c/source/test/intltest/itrbnf.cpp

```diff
@@ -2605,6 +2606,37 @@ IntlTestRBNF::TestNumberingSystem() {
     }
 }
 
+void
+IntlTestRBNF::TestInfiniteRecursion() {
+    UnicodeString badRules[] = {
+        ">>",
+        "<<",
+        "%foo: x=%foo=",
+        "%one: x>%two>; %two: y>%one>;"
+    };
+    
+    for (int32_t i = 0; i < UPRV_LENGTHOF(badRules); i++) {
+        UErrorCode err = U_ZERO_ERROR;
+        UParseError parseErr;
+        RuleBasedNumberFormat rbnf(badRules[i], parseErr, err);
+        
+        if (U_SUCCESS(err)) {
+            UnicodeString result;
+            rbnf.format(5, result);
+            // we don't actually care about the result and the function doesn't return an error code;
+            // we just want to make sure the function returns
+            
+            Formattable pResult;
+            rbnf.parse("foo", pResult, err);
+            assertTrue("rbnf.parse() didn't return U_INVALID_FORMAT_ERROR!", err = U_INVALID_FORMAT_ERROR);
```

Comment:
I think you meant `err == U_INVALID_FORMAT_ERROR` and not `err = U_INVALID_FORMAT_ERROR`

Reply:
(blush) Good catch!  Sorry about that...